Jenkinsfile

```
@@ -0,0 +1,259 @@
+                usernameColonPassword(credentialsId: 'Jenkins GitHub Account', variable: 'GIT_CREDENTIAL')]) {
+                    withDockerServer([uri: dockerHost()]) {
+                        withMaven(globalMavenSettingsConfig: 'jenkins-maven-global-settings', options: [findbugsPublisher(disabled: true)]) {
+                            // findbugs publishing is skipped in both steps because multi-module projects cause
+                            // extra copies to be reported. Instead, use commonPost to collect once at the end
+                            // of the build.
+                            writeFile file:'extract-iam-credential.sh', text:libraryResource('scripts/extract-iam-credential.sh')
+                            sh '''
+                                bash extract-iam-credential.sh
+
+                                # Hide login credential from below
+                                set +x
+
+                                LOGIN_CMD=$(aws ecr get-login --no-include-email --region us-west-2)
+
+                                $LOGIN_CMD
+                            '''
+                            sh '''
+                                echo $ARTIFACTORY_PASSWORD | docker login confluent-docker.jfrog.io -u $ARTIFACTORY_USERNAME --password-stdin
+                            '''
+                            writeFile file:'create-pip-conf-with-nexus.sh', text:libraryResource('scripts/create-pip-conf-with-nexus.sh')
+                            writeFile file:'create-pypirc-with-nexus.sh', text:libraryResource('scripts/create-pypirc-with-nexus.sh')
+                            writeFile file:'setup-credential-store.sh', text:libraryResource('scripts/setup-credential-store.sh')
+                            writeFile file:'set-global-user.sh', text:libraryResource('scripts/set-global-user.sh')
+                            sh '''
+                                bash create-pip-conf-with-nexus.sh
+                                bash create-pypirc-with-nexus.sh
+                                bash setup-credential-store.sh
+                                bash set-global-user.sh
+                            '''
+
+                            // Set the project versions in the pom files
+                            sh "mvn --batch-mode versions:set -DnewVersion=${config.ksql_db_version} -DgenerateBackupPoms=false"
+
+                            // Set the version of the parent project to use.
+                            sh "mvn --batch-mode versions:update-parent -DparentVersion=\"[${config.cp_version}]\" -DgenerateBackupPoms=false"
+
+                            if (config.release) {
+                                def git_tag = "v${config.ksql_db_version}-ksqldb"
+                                sh "git add ."
+                                sh "git commit -m \"Setting project version and parent version.\""
```

we have a plugin that requires commits to follow conventional commits, so I'm worried this might fail. let's be safe and change the commit message (also include the versions):
```suggestion
                                sh "git commit -m \"build: Setting project version ${config.ksql_db_version} and parent version ${config. cp_version}.\""
```